Userland/Libraries/LibVideo/VP9/Context.h

```diff
@@ -59,7 +59,7 @@ struct FrameContext {
 
     DecoderErrorOr<BooleanDecoder> create_range_decoder(size_t size)
     {
-        ReadonlyBytes stream_data = static_cast<FixedMemoryStream const&>(*stream).bytes();
+        ReadonlyBytes stream_data = stream->readonly_bytes();
```

Comment:
Does it actually make sense to reach inside a FixedMemoryStream to get its bytes? Having to know whether it's streaming from writable data or not feels awkward. That one use in the VP9 FrameContext could be avoided by just storing the ReadonlyBytes as another field.

`FixedMemoryStream::bytes()` (and `readonly_bytes()`) aren't used anywhere else, so I say get rid of them entirely.

Reply:
And it definitely feels like we are leaking some implementation detail. So, yes I agree that it should be removed down the road, but for now it allows a smooth transition from reading the old state which reads from a `Bytes` object, where the size is known.
This commit comes from a PNG Decoder PR where the situation is really similar.

Reply:
Ah, I didn't notice it gets used in the second commit. :thinking:

Still, the TGALoadingContext could just store the ReadonlyBytes too when it's created?

Reply:
I would rather keep that than all the plumbing to store these `Bytes`. But I don't have a really strong opinion either.
So that's up to you

Reply:
OK. I'll merge as-is, and maybe play around with removing it myself to see how that looks.